metric_source/local/local_test.go

```diff
@@ -322,3 +324,47 @@ func TestLocal_IsConfigured(t *testing.T) {
 		So(actual, ShouldBeTrue)
 	})
 }
+
+func TestLocal_evalExpr(t *testing.T) {
+	Convey("When everything is correct, we don't return any error", t, func() {
+		expression, _, err := parser.ParseExpr(`seriesByTag('name=k8s.dev-cl1.kube_pod_status_ready', 'condition!=true', 'namespace=default', 'pod=~*')`)
+		So(err, ShouldBeNil)
+		res, err := evalExpr("target", expression, time.Now().Add(-1*time.Hour).Unix(), time.Now().Unix(), nil)
+		So(err, ShouldBeNil)
+		So(res, ShouldBeNil)
+	})
+
+	Convey("When get panic, it should return error", t, func() {
+		expression, _, _ := parser.ParseExpr(`;fg`)
+		res, err := evalExpr("target", expression, 0, 0, nil)
+		So(err.Error(), ShouldContainSubstring, "panic while evaluate target target: message: 'runtime error: invalid memory address or nil pointer dereference")
+		So(res, ShouldBeNil)
+	})
+
+	Convey("When no metrics, should not return error", t, func() {
+		expression, _, err := parser.ParseExpr(`alias( divideSeries( alias( sumSeries( exclude( groupByNode( OFD.Production.{ofd-api,ofd-front}.*.fns-service-client.v120.*.GetCashboxRegistrationInformationAsync.ResponseCode.*.Meter.Rate-15-min-Requests-per-s, 9, "sum" ), "Ok" ) ), "bad" ), alias( sumSeries( OFD.Production.{ofd-api,ofd-front}.*.fns-service-client.v120.*.GetCashboxRegistrationInformationAsync.ResponseCode.*.Meter.Rate-15-min-Requests-per-s ), "total" ) ), "Result" )`)
+		So(err, ShouldBeNil)
+		res, err := evalExpr("target", expression, time.Now().Add(-1*time.Hour).Unix(), time.Now().Unix(), make(map[parser.MetricRequest][]*types.MetricData))
+		So(err, ShouldBeNil)
+		So(res, ShouldBeNil)
+	})
+
+	// Do we need 2 tests below?
+
+	//Convey("When give unknown func, return it", t, func() {
```

Comment:
ooops

Reply:
So do we need 2 tests below?

Reply:
why not?

Reply:
Now I uncommented and fixed 1st test, but removed 2nd test, because there is no internal error on our version of carbonapi

Reply:
ok, let's do it